pipeline/event.go

```diff
@@ -297,6 +298,9 @@ func (p *eventPool) dump() string {
 		o := logger.Header("events")
 		for i := 0; i < p.freeEventsCount; i++ {
 			event := p.events[i]
+			if event == nil {
+				continue
```

Comment:
Let's print `nil` event. 
Otherwise eventPool [m, m1, m2] will be dumped as string: `m.String()\nm2.String()\n` while expecting `m.String()\nnil\nm2.String()\n`

Reply:
we could also log an error in case this happens

Reply:
> we could also log an error in case this happens

Is not an error, event may nil if pool is busy. Should we log this?

Reply:
okay I agree, it's not necessary here